apps/cli/src/utils.ts

```diff
@@ -59,6 +61,23 @@ export function returnLoggerMiddleware(fn: (...args: any[]) => any | Promise<any
   };
 }
 
+export const trustedSourcesValidatorMiddleware = (fn: (...args: any[]) => any | Promise<any>) => {
+  return async function (...args: any[]) {
+    const url = args[0];
```

Comment:
One thing to improve in a follow up PR would be adding some sort of time checks for this

Reply:
sorry, type checks hahahah

Reply:
```ts
export const trustedSourcesValidatorMiddleware = (fn: (...args: any[]) => any | Promise<any>) => {
  return async function (...args: any[]) {
    const urls = args.filter((arg) => typeof arg === 'string' && arg.startsWith('http'));
    if (!urls) {
      return fn(...args);
    }

    const trustedSources = getTrustedSources();
    if (!trustedSources) {
      return fn(...args);
    }

    for (const url of urls) {
      if (!trustedSources.some((source) => picomatch(source)(url))) {
        throw new Error(`The URL is from an untrusted source: ${url}`);
      }
    }

    return fn(...args);
  };
};
```

maybe something like this?

Reply:
Sure